### PR TITLE
PSTRESS-89 Create a script to autogenerate rocksdb options file

### PR DESCRIPTION
**Pull request by mohitj1988**
This commit adds a script which will auto-generate rocksdb server options file which can be used by pstress to start server with random server variables.

Script: generate_rocksdb_options.sh
Output Filename: rocksdb_options_80.txt

### CODE REVIEW

File: pstress/generate_rocksdb_options.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+# Extract all options, their default values into temp file
+$BASEDIR/bin/mysql --no-defaults -S$SOCKET -uroot -e "SHOW GLOBAL VARIABLES LIKE '%rocksdb%'" > $TEMP_FILE
+
+# Remove the first line from the temp file
+# Variable_name, Variable_value (we do not need the column name displayed in temp file)
+sed -i 1d $TEMP_FILE
+
+# List of rocksdb variables which must not be changed.
+EXCLUDED_LIST=( --loose-rocksdb_compact_cf --loose-rocksdb_default_cf_options --loose-rocksdb_update_cf_options --loose-rocksdb_delete_cf --loose-rocksdb_override_cf_options --loose-rocksdb_create_checkpoint --loose-rocksdb_create_temporary_checkpoint --loose-rocksdb_fault_injection_options --loose-rocksdb_read_free_rpl_tables --loose-rocksdb_persistent_cache_path --loose-rocksdb_strict_collation_exceptions --loose-rocksdb_tmpdir --loose-rocksdb_trace_block_cache_access --loose-rocksdb_trace_queries --loose-rocksdb_wal_dir --loose-rocksdb_wsenv_path --loose-rocksdb_datadir )
```
Comment (inikep) [inline pstress/generate_rocksdb_options.sh:61]: `rocksdb_default_cf_options` is one of most important options.
Using this option you can bypass MyRocks and provide options directly to RocksDB.
Please check https://github.com/facebook/mysql-5.6/wiki/my.cnf-tuning for some examples

File: pstress/generate_rocksdb_options.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    echo "$OPTION=0.50" >> $OUTPUT_FILE
+    echo "$OPTION=0.75" >> $OUTPUT_FILE
+    echo "$OPTION=1.0" >> $OUTPUT_FILE
+  elif [[ "$OPTION" == "--loose-rocksdb_info_log_level" ]]; then
+    echoit " > Adding possible values (debug_level, info_level, warn_level, error_level, fatal_level ) for option '${OPTION}' to the final list..."
+    echo "$OPTION=debug_level" >> $OUTPUT_FILE
+    echo "$OPTION=info_level" >> $OUTPUT_FILE
+    echo "$OPTION=warn_level" >> $OUTPUT_FILE
+    echo "$OPTION=error_level" >> $OUTPUT_FILE
+    echo "$OPTION=fatal_level" >> $OUTPUT_FILE
```
Comment (satya-bodapati) [inline pstress/generate_rocksdb_options.sh:111]: this option can be skipped from randomness?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: pstress/generate_rocksdb_options.sh (hunk 2 above)
Comment (mohitj1988) [reply pstress/generate_rocksdb_options.sh:111]: Yeah. The more info the error log has, the better it is.
Thanks for the suggestion.

**Review by inikep — APPROVED** [review]
LGTM

File: pstress/generate_rocksdb_options.sh (hunk 1 above)
Comment (mohitj1988) [reply pstress/generate_rocksdb_options.sh]: Fixed

**Pull request merged**